File: rust_luogu/rust_codeforces/cpp/luogu/old/P1068.cpp
/*@author    Allergy
 *@email     Allergy527@gmail.com
 *@workspace luogu\P1068.cpp
 *@data      2024/01/17 21:55:35
 */
#include<bits/stdc++.h>
#define ll long long

using namespace std;

struct all_sc {
    int id, sc;
}qwq[5270];

bool cmp(all_sc a, all_sc b) {
    if(a.sc != b.sc)return a.sc > b.sc;
    else return a.id < b.id;
}

void solve() {
    int n, m, i, j;
    memset(qwq, 0, sizeof(qwq));
    cin >> n >> m;
    m = floor(m * 1.5);
    for(int i = 0;i < n;++i)cin >> qwq[i].id >> qwq[i].sc;
    sort(qwq, qwq + n, cmp);
    while(m < n && qwq[m - 1].sc == qwq[m].sc)++m;
    cout << qwq[m - 1].sc << ' ' << m << '\n';
    for(int i = 0;i < m;++i)cout << qwq[i].id << ' ' << qwq[i].sc << '\n';
}

int main() {
    ios::sync_with_stdio(false);
    int t = 1;
    //cin >> t;
    while(t--)solve();
    cin >> t;
    return 0;
}

File: rust_luogu/src/bin/Part 2.2/p1068.rs
//@author    Allergy
//@email     Allergy527@gmail.com
//@workspace luogu/p1068.rs
//@data      2024/01/20 15:27:40
fn cin() -> String {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    input.trim().to_string()
}
struct St {
    id: usize,
    score: usize,
}

fn main() {
    let t = 1;
    //let mut input = String::new();std::io::stdin().read_line(&mut input).unwrap();let t:i32=input.parse().unwrap();
    (0..t).for_each(|_| solve());
}
fn solve() {
    let (a, b) = cin()
        .split_whitespace()
        .fold((0, 0), |x, y| (x.1, y.parse::<usize>().unwrap()));
    let mut b = (b * 3).div_euclid(2);
    let mut qwq = Vec::new();
    for _ in 0..a {
        let (p, q) = cin()
            .split_whitespace()
            .fold((0, 0), |x, y| (x.1, y.parse::<usize>().unwrap()));
        qwq.push(St { id: p, score: q })
    }
    qwq.sort_unstable_by(|y, x| {
        if x.score != y.score {
            x.score.cmp(&y.score)
        } else {
            y.id.cmp(&x.id)
        }
    });
    while qwq[b + 1].score == qwq[b].score {
        b += 1
    }
    println!("{} {}", qwq[b - 1].score, b+1);
    qwq[0..=b]
        .iter()
        .for_each(|x| println!("{} {}", x.id, x.score))
}
